src/debug_sdfs_ed64.c

```diff
@@ -159,6 +159,8 @@ static bool everdrive_sd_read_response(uint8_t res_buff[5]) {
 		res_buff[2] = everdrive_sd_read_command();
 		res_buff[3] = everdrive_sd_read_command();
 		res_buff[4] = everdrive_sd_read_command();
+	} else {
+		timeout += 4;
```

Comment:
Why is this required?

Reply:
To make sure we consume the data from the sd card even if we don't use it.

Reply:
Obviously the semantic is a bit different. 

 * If you pass res_buff != NULL, you always read 4 bytes (with no checks), and then consume up to 16 bytes, stopping at the first 0xFF.
 * If you pass res_buff==NULL, you consume up to 20 bytes stopping at the first 0xFF.

I don't know if it matters. If passing res_buff==NULL is just a way for the caller to avoid reading a data that is not useful, then maybe we should do something like:

```C
for (int i=0;i<4;i++) {
    uint8_t data = everdrive_sd_read_command();
    if (res_buff) res_buff[i+1] = data;
}
```

which would preserve the semantic of checking for 0xFF just in the second loop. But again, I don't understand this protocol very well.

Reply:
This is a combination of possible sized responses (depending on the command we can have up to 20 bytes), it makes it a little less ergonomic but simpler. The clock line is only toggled when we actually read/write so we always need to complete the data flow specified by the protocol to keep things as expected b/c we are the host device and we decide when to continue transfers.

That's a nice implementation suggestion as well I will do that instead.